crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -38,10 +38,10 @@ enum StateTransition<T: Component + Clone + Eq> {
 
 #[derive(Debug)]
 enum ScheduledOperation<T: Component + Clone + Eq> {
-    Set(T),
-    Replace(T),
-    Pop,
-    Push(T),
+    Set(T, bool),
+    Replace(T, bool),
+    Pop(bool),
+    Push(T, bool),
```

Comment:
This is a bit of a design nit-pick, but generally I don't like using `bool` when an `enum` can be far more descriptive.
For example:
```rust
let result = state.set(MyState::Hello, true);
```
is not simple to understand what `true` is doing here. 

However if we had a 
```rust
enum ReRun { // naming could use work lol, uwu
    Yes, No,
}
```
and then 
```rust
let result = state.set(MyState::Hello, ReRun::Yes);
```
This variant is far clearer and much more descriptive.

Reply:
changed the api to be better typed and non-breaking 👍 
`state.set(AppState::Game).unwrap().next(Next::NextFrame);`